java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/client/utils/ClientAuthenticationUtilsTest.java

```diff
@@ -169,13 +169,15 @@ private MockedStatic<ClientAuthenticationUtils> createClientAuthenticationUtilsS
       if (method.getName().equals("getCertificateInputStreamFromSystem")) {
         return invocationOnMock.callRealMethod();
       }
-      return invocationOnMock.getMock();
+      return method.invoke(invocationOnMock.getMock(), invocationOnMock.getArguments());
     });
   }
 
   private void setOperatingSystemMock(MockedStatic<ClientAuthenticationUtils> clientAuthenticationUtilsMockedStatic,
                                       boolean isWindows, boolean isMac) {
     clientAuthenticationUtilsMockedStatic.when(ClientAuthenticationUtils::isMac).thenReturn(isMac);
+    Assert.assertEquals(ClientAuthenticationUtils.isMac(), isMac);
```

Comment:
I can leave this out of the final diff if requested, but I think having this check could have caught the test suite bug in prior versions.